sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/EquivalentExpressions.scala

```diff
@@ -147,7 +147,7 @@ class EquivalentExpressions {
       expr.find(_.isInstanceOf[LambdaVariable]).isDefined ||
       // `PlanExpression` wraps query plan. To compare query plans of `PlanExpression` on executor,
       // can cause error like NPE.
-      (expr.isInstanceOf[PlanExpression[_]] && TaskContext.get != null)
+      (expr.find(_.isInstanceOf[PlanExpression[_]]).isDefined && TaskContext.get != null)
```

Comment:
I think we should fix it at the master branch as well, as the code does not match the comment.

We can also add a UT in `SubexpressionEliminationSuite`, which tests `addExprTree` directly.

Reply:
OK, i will open another pr to fix it at the master branch and add a UT

Reply:
> We can also add a UT in `SubexpressionEliminationSuite`, which tests `addExprTree` directly.

I think add a UT in SubexpressionEliminationSuite is not reasonable, as we can not access InSubqueryExec in Spark Catalyst module

Reply:
unit test added

Reply:
@cloud-fan  How should this pr title be named? Maybe this is a potential problem.

Reply:
Do we have a master branch PR now?

Reply:
@monkeyboy123 we can replace `expr.find(_.isInstanceOf[PlanExpression[_]]).isDefined` with `TreeNode.exists` api now

Reply:
> Do we have a master branch PR now?
Sorry for late reply，i will open a pr right now.

Reply:
> @monkeyboy123 we can replace `expr.find(_.isInstanceOf[PlanExpression[_]]).isDefined` with `TreeNode.exists` api now
ok

Reply:
> potential

done, new pr [SPARK-38333](https://github.com/apache/spark/pull/36012)